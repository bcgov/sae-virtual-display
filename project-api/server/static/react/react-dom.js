Cannot find "/dist/react-dom.js" in react@0.14.3