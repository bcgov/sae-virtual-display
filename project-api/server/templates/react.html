<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" type="text/css" href="/static/bootstrap/bootstrap.min.css"/>
        <script src="/static/bootstrap/jquery-3.4.1.min.js"></script>
        <script src="/static/bootstrap/bootstrap.min.js"></script>
        <title>DIP Self-Service</title>
    </head>
    <body>
            <div class="container">
                    <h1>Groups</h1>
                    <p>{{ groups }}</p>
                    <div id="root"></div>
                    <script src="/static/react/react.min.js"></script>
                    <script src="/static/react/react-dom.min.js"></script>
                    <script src="/static/react/browser.min.js"></script>
            
                    <script type="text/babel" src="/static/app/greeting.jsx"></script>
            </div>
    </body>
</html>
