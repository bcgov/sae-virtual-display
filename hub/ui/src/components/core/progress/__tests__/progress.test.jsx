import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import Progress from '../';

describe('core/progress', () => {
  afterEach(cleanup);

  it('should render, with defaults', () => {
    const { container, getByRole, getByText } = render(<Progress />);
    expect(getByRole('progressbar')).toHaveStyle('width: 0%');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a message', () => {
    const message = 'This is loading';
    const { getByText } = render(<Progress message={message} />);
    expect(getByText(message)).toBeTruthy();
  });

  it('should render the progress bar correctly', () => {
    const { getByRole } = render(<Progress value={40} />);
    expect(getByRole('progressbar')).toHaveStyle('width: 40%');
    expect(getByRole('progressbar')).toHaveAttribute('aria-valuenow', '0.4');
  });

  it('should not render values below 0', () => {
    const { getByRole } = render(<Progress value={-5} />);
    expect(getByRole('progressbar')).toHaveStyle('width: 0%');
    expect(getByRole('progressbar')).toHaveAttribute('aria-valuenow', '0');
  });
});
