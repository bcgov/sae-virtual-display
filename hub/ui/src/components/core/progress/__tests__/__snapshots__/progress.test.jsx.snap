// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/progress should render, with defaults 1`] = `
.c0 {
  text-align: left;
}

.c2 {
  margin: 15px 0 0;
}

.c1 {
  font-size: 1em;
  font-style: inherit;
  line-height: 1.1428571428571428;
  color: #172B4D;
  font-weight: 600;
  -webkit-letter-spacing: -0.003em;
  -moz-letter-spacing: -0.003em;
  -ms-letter-spacing: -0.003em;
  letter-spacing: -0.003em;
  margin-top: 16px;
  margin-bottom: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 0.7857142857142857em;
  font-style: inherit;
  line-height: 1.4545454545454546;
  color: #6B778C;
  font-weight: 700;
  margin-top: 16px;
  font-weight: 500;
  margin-top: 0.5rem;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Request in progress
  </p>
  <div
    class="c2"
  >
    <div
      aria-valuemax="1"
      aria-valuemin="0"
      aria-valuenow="0"
      class="css-1656qb0"
      role="progressbar"
      tabindex="0"
    >
      <span
        class="css-1i5tcqv-Bar"
      />
      <span
        class="css-1p25adh-Bar"
      />
    </div>
  </div>
  <p
    class="c3"
  >
    <span>
      Processing...
    </span>
    <span>
      0% complete
    </span>
  </p>
</div>
`;
